swebench/harness/test_spec/test_spec.py

```diff
@@ -203,13 +203,10 @@ def _from_json_or_obj(key: str) -> Any:
     env_name = "testbed"
     repo_directory = f"/{env_name}"
 
-    if MAP_REPO_VERSION_TO_SPECS.get(repo, None):
-        specs = MAP_REPO_VERSION_TO_SPECS[repo][version]
-    elif instance.get("install", None):
-        specs = instance["install"]
-        if isinstance(specs, str):
-            specs = ast.literal_eval(specs)
-
+    specs = MAP_REPO_VERSION_TO_SPECS.get(repo, instance.get("install", None))
+    if isinstance(specs, str):
+        specs = json.loads(specs)
+    specs = specs.get(version, specs)
```

Comment:
This is quite hacky. Why not simply do:
```
if repo in MAP_REPO_VERSION_TO_SPECS:
   specs = MAP_REPO_VERSION_TO_SPECS[repo][version]
else:
   specs = json.loads(instance["install"])
```

According to what you told me, this should always work and it will throw an error if we get unexpected inputs

Reply:
Then put it and test it :)